### PR TITLE
macos route poking

### PR DESCRIPTION
**Pull request by majestrate**
this makes route poking work on macos

### CODE REVIEW

File: llarp/net/route.cpp [hunk 1]
```diff
@@ -399,35 +399,36 @@ namespace llarp::net
...
+      for (auto part : parts)
       {
-        line_str = line_str.substr(7);
-        while (line_str[0] == ' ')
-        {
-          line_str = line_str.substr(1);
-        }
-        const auto pos = line_str.find(" ");
-        if (pos != std::string::npos)
+        if (not part.empty())
```
Comment (tewinget) [inline llarp/net/route.cpp:423]: It seems that if you use `split_any(input_str, delimiter(s), bool trim)` with trim=true then you will get this behavior without having to check here for empty splits.

**Comment by majestrate** [comment]
The documentation is unclear if that really does yield the
desired result.

File: llarp/router/router.cpp [hunk 2]
```diff
@@ -770,6 +770,47 @@ namespace llarp
...
+        for (const auto& [ip, gw] : m_PokedRoutes)
+        {
+          // save hole
+          holes.emplace(ip);
+          // unpoke route
+          net::DelRoute(ip, gw);
+        }
+        m_PokedRoutes.clear();
+
+        if (not gateway.empty())
```
Comment (tewinget) [inline llarp/router/router.cpp:803]: Seems this is basically "our gateway changed, so check if that means we've just lost internet connectivity entirely".  In that case, will the routes we removed above be recreated properly when connectivity is restored?  Or will new first hops be selected rendering that moot?

File: llarp/service/endpoint.cpp [hunk 3]
```diff
@@ -914,7 +915,24 @@ namespace llarp
         RemoveConvoTag(frame.T);
         return true;
       }
-      return frame.AsyncDecryptAndVerify(EndpointLogic(), p, m_Identity, this);
+      if (not frame.AsyncDecryptAndVerify(EndpointLogic(), p, m_Identity, this))
+      {
```
Comment (tewinget) [inline llarp/service/endpoint.cpp:919]: Is this (and should it be) doing crypto in the logic thread?

File: llarp/service/endpoint.cpp [hunk 4]
```diff
@@ -914,7 +915,24 @@ namespace llarp
...
+        // send reset convo tag message
+        ProtocolFrame f;
+        f.R = 1;
+        f.T = frame.T;
+        f.F = p->intro.pathID;
+
+        f.Sign(m_Identity);
```
Comment (tewinget) [inline llarp/service/endpoint.cpp:926]: Is this (and should it be) doing crypto in the logic thread?

Comment (majestrate) [reply llarp/service/endpoint.cpp:926]: not frequent enough to warrant it imo

File: llarp/router/router.cpp (hunk 2 above)
Comment (majestrate) [reply llarp/router/router.cpp:803]: yes they will be recreated later on demand.

File: llarp/service/endpoint.cpp (hunk 4 above)
Comment (tewinget) [reply llarp/service/endpoint.cpp:926]: ok, just checking

File: llarp/net/route.cpp (hunk 1 above)
Comment (majestrate) [reply llarp/net/route.cpp:423]: meh

Comment (majestrate) [reply llarp/net/route.cpp]: split_any used

**Review by tewinget — APPROVED** [review]

**Pull request merged**